src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,148 @@
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void,
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  // eslint-disable-next-line max-len
+  const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+  const [movie, setMovie] = useState({
+    title: {
+      value: '',
+      isRequired: true,
+    },
+    description: {
+      value: '',
+      isRequired: false,
+    },
+    imgUrl: {
+      value: '',
+      isRequired: true,
+    },
+    imdbUrl: {
+      value: '',
+      isRequired: true,
+    },
+    imdbId: {
+      value: '',
+      isRequired: true,
+    },
+  });
+
+  const [count, setCount] = useState(0);
+  const hasErrors = (movie.title.isRequired && !movie.title.value)
+    || (movie.description.isRequired && !movie.description.value)
+    || (movie.imgUrl.isRequired && !movie.imgUrl.value)
+    || (movie.imdbUrl.isRequired && !movie.imdbUrl.value)
+    || (movie.imdbId.isRequired && !movie.imdbId.value)
+    || !pattern.test(movie.imgUrl.value)
+    || !pattern.test(movie.imdbUrl.value);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+
+    setMovie((prevMovie) => ({
+      ...prevMovie,
+      [name]: {
+        ...prevMovie[name as keyof typeof prevMovie],
+        value,
+      },
+    }));
+  };
+
+  const resetForm = () => {
+    setMovie(prevMovie => ({
+      title: {
+        ...prevMovie.title,
+        value: '',
+      },
+      description: {
+        ...prevMovie.description,
+        value: '',
+      },
+      imgUrl: {
+        ...prevMovie.imgUrl,
+        value: '',
+      },
+      imdbUrl: {
+        ...prevMovie.imdbUrl,
+        value: '',
+      },
+      imdbId: {
+        ...prevMovie.imdbId,
+        value: '',
+      },
+    }));
+  };
+
+  const onFormSubmit = (event: React.FormEvent) => {
+    const newMovie: Movie = {
+      title: movie.title.value,
+      description: movie.description.value,
+      imgUrl: movie.imgUrl.value,
+      imdbUrl: movie.imdbUrl.value,
+      imdbId: movie.imdbId.value,
+    };
+
+    event.preventDefault();
+    setCount(prev => prev + 1);
+    onAdd(newMovie);
+    resetForm();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={onFormSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
-        required
+        value={movie.title.value}
+        onChange={handleChange}
+        required={movie.title.isRequired}
```

Comment:
> no idea if it's a good approach to store isRequired flag in movie UseState :)

good point :slightly_smiling_face: - it's not
You never change it - so there is no reason to store it in state (no need to store it anywhere - just pass boolean prop to `TextField` if it's required or not - if it's not)

```suggestion
        required
```

Reply:
That's what i did first but then i stuck trying to make my submit button disabled if any field has an error, because description is not required like other fields and i can't access it outside of component